src/height/service.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  IMerjoonProjects,
+  IMerjoonService,
+  IMerjoonTasks,
+  IMerjoonUsers,
+} from "../common/types";
+import { HeightApi } from "./api";
+import { HeightTransformer } from "./transformer";
+import {
+  HeightApiPath,
+  IHeightGeneralType,
+  IHeightList,
+  IHeightQueryParams,
+  IHeightTask,
+  IHeightUser,
+} from "./types";
+
+export class HeightService implements IMerjoonService {
+  constructor(
+    public readonly api: HeightApi,
+    public readonly transformer: HeightTransformer
+  ) {}
+
+  protected async *getAllRecordsIterator(
+    path: HeightApiPath,
+    queryParams?: IHeightQueryParams
+  ) {
+    const { limit = 50 } = queryParams ?? {};
+    let shouldStop = false;
+    let lastRetrievedDate: string | null = null;
+
+    do {
+      try {
+        // filters apply only to tasks
+        // filters tasks created before lastRetrievedDate
+        const filters = lastRetrievedDate
```

Comment:
not a good idea to have a generic function, and put code which is specific only to one use case. let's come up with other idea here